lib/base.php

```diff
@@ -18,29 +18,45 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+use Monolog\Logger;
+use Monolog\ErrorHandler;
+use Monolog\Handler\ErrorLogHandler;
+use Monolog\Handler\StreamHandler;
+
 class Mesamatrix
 {
     public static $serverRoot; // Path to root of installation
     public static $configDir; // Path to configuration directory
     public static $config; // Config object
     public static $autoloader; // Autoloader
+    public static $logger; // Logger
 
     public static function init() {
+        date_default_timezone_set('UTC');
+
         $dir = str_replace("\\", '/', __DIR__);
         self::$serverRoot = implode('/', array_slice(explode('/', $dir), 0, -1));
 
-        self::$autoloader = (require self::$serverRoot.'/vendor/autoload.php');
+        self::$autoloader = (require self::path('vendor/autoload.php'));
 
-        self::$configDir = self::$serverRoot.'/config';
+        self::$logger = new Logger('logger');
+        self::$logger->pushHandler(new ErrorLogHandler(
+            ErrorLogHandler::OPERATING_SYSTEM,
+            Logger::NOTICE
+        ));
+        ErrorHandler::register(self::$logger);
+
+        self::$configDir = self::path('config');
         self::$config = new \Mesamatrix\Config(self::$configDir);
 
-        date_default_timezone_set('UTC');
-    }
+        // register the log file
+        self::$logger->popHandler();
+        self::$logger->pushHandler(new StreamHandler(
+            self::path(self::$config->getValue('info', 'private_dir').'/mesamatrix.log'),
+            self::$config->getValue('info', 'log_level', Logger::WARNING)
+        ));
```

Comment:
From what I understand, logs will be redirected only to a file. Maybe it would be interesting to have both logs, one stored in a file and one live in the terminal?